homeassistant/components/somfy/__init__.py

```diff
@@ -92,6 +93,9 @@ async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry):
     async def _update_all_devices():
         """Update all the devices."""
         devices = await hass.async_add_executor_job(data[API].get_devices)
+        # Sometimes Somfy returns an empty list.
+        if not devices:
+            raise UpdateFailed("No devices returned.")
```

Comment:
Shouldn't this be handled by the library upstream? (e.g., by throwing an error in those cases?)
And is it possible to have a valid connection without any devices? If so, should it only trigger if it goes from 1 or more devices to zero?

Reply:
@frenck The upstream library cannot detect this case. Somfy returns a correct answer: same response format but empty with status code 200.
Your 2nd point make sense, I can add a check about the previous number of devices. And raise an error if we go from any  to 0.